tests/core/editor/custom_config_race.js

```diff
@@ -1,41 +1,66 @@
 /* bender-tags: editor */
 
-var assetsPath = '%TEST_DIR%_assets/',
-	instances = {},
-	initEditors = ( function() {
-		var count = 0;
-
-		return function( editors, callback ) {
-			var total = CKEDITOR.tools.object.keys( editors ).length;
-
-			for ( var e in editors ) {
-				instances[ e ] = CKEDITOR.replace( e,
-					CKEDITOR.tools.extend( editors[ e ], {
-						plugins: 'wysiwygarea',
-						on: {
-							instanceReady: function() {
-								if ( ++count == total )
-									callback();
+var assetsPath = '%TEST_DIR%_assets/';
+
+function initEditors( editors ) {
+	var count = 0;
+	var instances = {};
+
+	return function( callback ) {
+		var total = CKEDITOR.tools.object.keys( editors ).length;
+
+		for ( var e in editors ) {
+			instances[ e ] = CKEDITOR.replace( e,
+				CKEDITOR.tools.extend( editors[ e ], {
+					plugins: 'wysiwygarea',
+					on: {
+						instanceReady: function() {
+							if ( ++count == total ) {
+								resume( function() {
+									callback( instances );
+								} );
 							}
 						}
-					} )
-				);
-			}
-		};
-	} )();
+					}
+				} )
+			);
+		}
+
+		wait();
+	};
+}
 
 bender.test( {
-	'async:init': function() {
+	tearDown: function() {
+		for ( var editorName in CKEDITOR.instances ) {
+			CKEDITOR.instances[ editorName ].destroy();
+		}
+	},
+
+	// (#3361)
+	'test race of customConfigs with reused config': function() {
 		initEditors( {
 			editor1: { customConfig: assetsPath + 'raceconfig1.js' },
 			editor2: { customConfig: assetsPath + 'raceconfig2.js' },
-			editor3: { customConfig: assetsPath + 'raceconfig3.js' }
-		}, this.callback );
+			editor3: { customConfig: assetsPath + 'raceconfig1.js' }
+		} )
+		( function( instances ) {
+			assert.areSame( '200px', instances.editor1.config.width, 'Instance uses own customConfig.' );
+			assert.areSame( '300px', instances.editor2.config.width, 'Instance uses own customConfig.' );
+			assert.areSame( '200px', instances.editor3.config.width, 'Instance uses own customConfig.' );
```

Comment:
AFAIK the last parameter here is an error message, so something like `Instance should use own customConfig.` or `Instance didn't use own customConfig.` could be a little more precise communicate.